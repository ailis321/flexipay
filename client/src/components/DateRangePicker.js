import React from "react";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

const DateRangePicker = ({ startDate, endDate, setStartDate, setEndDate }) => {
  return (
    <div>
      <DatePicker
        id="start-date-picker"
        selected={startDate}
        onChange={setStartDate}
        selectsStart
        startDate={startDate}
        endDate={endDate}
        dateFormat="d MMMM, yyyy"
        data-testid="start-date-picker"
      />
      <DatePicker
        id="end-date-picker"
        selected={endDate}
        onChange={setEndDate}
        selectsEnd
        startDate={startDate}
        endDate={endDate}
        minDate={startDate}
        dateFormat="d MMMM, yyyy"
        data-testid="end-date-picker"
      />
    </div>
  );
};

export default DateRangePicker;
