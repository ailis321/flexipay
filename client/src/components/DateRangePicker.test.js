import React from 'react';
import { render } from '@testing-library/react';
import DateRangePicker from './DateRangePicker';

describe('DateRangePicker', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <DateRangePicker
        startDate={new Date()}
        endDate={new Date()}
        setStartDate={() => {}}
        setEndDate={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
