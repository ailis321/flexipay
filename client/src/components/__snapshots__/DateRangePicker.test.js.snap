// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          class=""
          type="text"
          value="5 May, 2024"
        />
      </div>
    </div>
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          class=""
          type="text"
          value="5 May, 2024"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
